src/Libraries/CoreNodeModels/CustomSelectionItem.cs

```diff
@@ -0,0 +1,165 @@
+using System;
+using System.ComponentModel;
+using System.Windows.Input;
+using Newtonsoft.Json;
+
+namespace CoreNodeModels
+{
+    /// <summary>
+    /// Represents an Enum item
+    /// </summary>
+    public class CustomSelectionItem
+    {
+        public string Name { get; set; }
+
+        public string Value { get; set; }
+
+        public CustomSelectionItem()
+        {
+
+        }
+    }
+
+
+    public class CustomSelectionItemViewModel : INotifyPropertyChanged, IDataErrorInfo
+    {
+        private readonly CustomSelectionItem customSelectionItem;
+
+        public event PropertyChangedEventHandler PropertyChanged;
+
+        #region Properties
+
+        [JsonIgnore]
+        public Func<CustomSelectionItem, bool> IsUnique { get; set; }
+
```

Comment:
definitely document these public properties, and try to reduce their access if possible.